### PR TITLE
Service release process step 1 to 4 added

### PR DESCRIPTION
**Pull request by manojava-gk**

### CODE REVIEW

File: cx-portal/src/assets/locales/de/servicerelease.json [hunk 1]
```diff
@@ -12,58 +12,42 @@
...
-    "defaultCardAppImageAlt": "Part Chain",
-    "defaultCardAppProvider": "Catena-X",
-    "defaultCardAppTitle": "This is the app title",
-    "defaultCardShortDescriptionEN": "[short description EN] this section will get replaced with your app short description",
-    "appTitle": "App Title",
-    "appProvider": "App Provider",
+  "step1": {
+    "headerTitle": "Create Service",
+    "headerDescription": "The following input fields and uploads are used to create your service 'Service Card' which will be placed inside the marketplace. The service card is the first touchpoint with the customer. While entering the data below, have a look at the service card on the left side which is getting autoupdated based on your input.",
+    "defaultCardServiceImageAlt": "Part Chain",
```
Comment (jjeroch) [inline cx-portal/src/assets/locales/de/servicerelease.json:23]: Lets not take real data - please change to "{Service Name}" - same applies for Provider field

File: cx-portal/src/assets/locales/en/servicerelease.json [hunk 2]
```diff
@@ -12,58 +12,42 @@
...
   },
-  "appPage": {
-    "headerTitle": "App Details",
+  "step2": {
+    "headerTitle": "Service Details",
     "headerDescription": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard .Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard .",
     "longDescriptionEN": "Long Description (en)",
     "longDescriptionDE": "Long Description (de)",
-    "images": "Images",
+    "images": "Image",
```
Comment (jjeroch) [inline cx-portal/src/assets/locales/en/servicerelease.json:38]: Please recheck, if thats the image part in the first step of the form; please use "Service Card Image" to ensure that its more clear for the user whats actually required/requested

File: cx-portal/src/assets/locales/en/servicerelease.json [hunk 3]
```diff
@@ -12,58 +12,42 @@
...
-    "appLeadImageUpload": "App Lead Image Upload"
+    "serviceLeadImageUpload": "Service Lead Image Upload"
   },
-  "appPage": {
-    "headerTitle": "App Details",
+  "step2": {
+    "headerTitle": "Service Details",
     "headerDescription": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard .Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard .",
     "longDescriptionEN": "Long Description (en)",
     "longDescriptionDE": "Long Description (de)",
```
Comment (jjeroch) [inline cx-portal/src/assets/locales/en/servicerelease.json:37]: I am missing in the locale files the "Supporting Documents" as per the jira ticket

File: cx-portal/src/components/pages/ServiceReleaseProcess/components/index.tsx [hunk 4]
```diff
@@ -61,17 +57,14 @@ export default function ServiceReleaseProcessForm() {
...
-      pageHeaderTitle={t('content.apprelease.headerTitle')}
-      headerTitle={t('content.apprelease.submitApp.headerTitle')}
-      headerDescription={t('content.apprelease.submitApp.headerDescription')}
-      headerDescriptionComplete={t(
-        'content.apprelease.submitApp.headerDescriptionComplete'
-      )}
-      yourCatenaXTeam={t('content.apprelease.submitApp.yourCatenaXTeam')}
-      myAppsOverview={t('content.apprelease.submitApp.myAppsOverview')}
+      pageHeaderTitle={t('headerTitle')}
+      headerTitle={t('submitApp.headerTitle')}
```
Comment (jjeroch) [inline cx-portal/src/components/pages/ServiceReleaseProcess/components/index.tsx:63]: @manojava-gk please recheck the section - does it make sense to call it "APP" in row 62 to 67

**Review by jjeroch — CHANGES_REQUESTED** [review]

File: cx-portal/src/assets/locales/en/servicerelease.json [hunk 5]
```diff
@@ -115,19 +99,20 @@
...
+    "myAppsOverview": "My services overview"
   },
-  "appReleaseForm": {
+  "serviceReleaseForm": {
     "headerTitle": "Create App",
     "addMore": "Add More",
     "noItemsSelected": "No Items Selected",
     "note": "Note:",
     "OnlyOneFileAllowed": "Only one file for upload allowed",
     "max3Images": "Max. 3 Images are supported",
```
[the remark this answers is not in the document]
Comment (manojava-gk) [reply cx-portal/src/assets/locales/en/servicerelease.json:114]: This is meant for service release section. I have copied everything from appRelease section. Will remove unwanted key value pair

File: cx-portal/src/components/pages/ServiceReleaseProcess/components/index.tsx (hunk 4 above)
Comment (manojava-gk) [reply cx-portal/src/components/pages/ServiceReleaseProcess/components/index.tsx:63]: These are some keys that has been copied from main.json. Noted. Will update this

File: cx-portal/src/assets/locales/en/servicerelease.json (hunk 3 above)
Comment (manojava-gk) [reply cx-portal/src/assets/locales/en/servicerelease.json:37]: Noted

**Comment by manojava-gk** [comment]
@jjeroch Could you pls review this again?

File: cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 6]
```diff
@@ -349,81 +349,61 @@ export default function AppPage() {
...
+                  )} ${
+                    item === 'longDescriptionEN'
+                      ? `a-zA-Z0-9 !?@&#'"()[]_-+=<>/*.,;:`
+                      : `a-zA-ZÀ-ÿ0-9 !?@&#'"()[]_-+=<>/*.,;:`
+                  }`,
+                  maxLength: `${t(
+                    'content.apprelease.appReleaseForm.maximum'
+                  )} 255 ${t(
+                    'content.apprelease.appReleaseForm.charactersAllowed'
+                  )}`,
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:404]: this doesn't make any sense - reading configuration values from language files?

File: cx-portal/src/components/shared/basic/ReleaseProcess/ContractAndConsent/index.tsx [hunk 7]
```diff
@@ -109,19 +105,96 @@ export default function ContractAndConsent() {
...
+          defaultuploadImageConformity &&
+          defaultuploadImageConformity[0]?.documentId,
+        name:
+          defaultuploadImageConformity &&
+          defaultuploadImageConformity[0]?.documentName,
+        status: UploadStatus.UPLOAD_SUCCESS,
+      })
+      setFileStatus('uploadImageConformity', UploadStatus.UPLOAD_SUCCESS)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/ContractAndConsent/index.tsx:146]: we need to stop this practice disabling eslinter rules in useEffect

File: cx-portal/src/components/shared/basic/ReleaseProcess/ContractAndConsent/index.tsx [hunk 8]
```diff
@@ -109,19 +105,96 @@ export default function ContractAndConsent() {
...
+
+      uploadDocumentApi(appId, 'CONFORMITY_APPROVAL_BUSINESS_APPS', value)
+        .then(() =>
+          setFileStatus('uploadImageConformity', UploadStatus.UPLOAD_SUCCESS)
+        )
+        .catch(() =>
+          setFileStatus('uploadImageConformity', UploadStatus.UPLOAD_ERROR)
+        )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/ContractAndConsent/index.tsx:190]: we need to stop this practice disabling eslinter rules in useEffect

File: cx-portal/src/components/shared/basic/ReleaseProcess/OfferCard/index.tsx [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+      serviceStatusDate?.documents?.APP_LEADIMAGE &&
+      serviceStatusDate?.documents?.APP_LEADIMAGE[0].documentId
+    ) {
+      fetchCardImage(
+        serviceStatusDate?.documents?.APP_LEADIMAGE[0].documentId,
+        serviceStatusDate?.documents?.APP_LEADIMAGE[0].documentName
+      )
+    }
+    reset(defaultValues)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/OfferCard/index.tsx:124]: we need to stop this practice disabling eslinter rules in useEffect

File: cx-portal/src/components/shared/basic/ReleaseProcess/OfferPage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+          status,
+        }
+        setValue('images', files)
+      }
+
+      for (let fileIndex = 0; fileIndex < imgs.length; fileIndex++) {
+        setFiles(fileIndex, UploadStatus.UPLOAD_SUCCESS)
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (oyo) [inline cx-portal/src/components/shared/basic/ReleaseProcess/OfferPage/index.tsx:117]: we need to stop this practice disabling eslinter rules in useEffect

**Review by oyo — CHANGES_REQUESTED** [review]
Disabling eslinter rules can be acceptable in some rare cases but it's now becoming a common pattern in your code. Also it's no good practice to implement a lot of logic in a chain of useEffects. Folders should never be called `CommonFiles` and services not `CommonService` because it doesn't say anything about the purpose of the code in it.

**Comment by manojava-gk** [comment]
@oyo As discussed with you over the call, multiple useEffects and eslinter disable code is not added in this PR. I have just created a wrapper component and moved all files to a specific folder.

**Review by oyo — APPROVED** [review]
as discussed approved for now because most of it was existing code. We are going to address the issues with a follow up ticket

**Review by jjeroch — APPROVED** [review]

**Pull request merged**